llvm/lib/SYCLLowerIR/LowerESIMDVecArg.cpp

```diff
@@ -181,8 +181,12 @@ Function *ESIMDLowerVecArgPass::rewriteFunc(Function &F) {
 
   llvm::CloneFunctionInto(NF, &F, VMap, F.getSubprogram() != nullptr, Returns);
 
+  // insert bitcasts in new function only if its a definition
   for (auto &B : BitCasts) {
-    NF->begin()->getInstList().push_front(B);
+    if (!F.isDeclaration())
```

Comment:
why not to move this check to the loop above and avoid creation and addition to `BitCasts` in case F is declaration? Am I missing something? Also if F is a declaration, do we need to clone, or just creating a new declaration is enough?

Reply:
i'm not sure whether CloneFunctionInto is needed for declaration as well, i assumed it is. since it uses VMap and given that VMap is populated in the loop above it, i chose to use this fix.

Reply:
I suggest to check this and remove redundant code if cloning is not really needed.

Reply:
it is needed as it adds attribute on the function declaration (dso_local).

Reply:
Attribute can be added w/o using cloning, right?

Reply:
i prefer using existing API as it works for both definitions/declarations. in case llvm adds more fields for functions, this API would ensure automatic compliance.